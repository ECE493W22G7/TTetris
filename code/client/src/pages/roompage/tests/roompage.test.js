import React from 'react';
import '@testing-library/jest-dom';
import MockedSocket from 'socket.io-mock';
import RoomPage from 'pages/roompage/roompage';
import { Routes, Route, MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import GameViewComponent from '../components/gameView/gameViewComponent';
import GameSettingsComponent from '../components/gameSettingsComponent';

describe("Testing Room Page", () => {
    let socket;

    beforeEach(() => {
        socket = new MockedSocket();
    })

    afterEach(() => {
        jest.restoreAllMocks();
    })

    test("Should show up", () => {
        const tree = renderer
            .create(<MemoryRouter>
                <Routes>
                    <Route path="/" element={<RoomPage socket={socket.socketClient}/>} />
                </Routes>
            </MemoryRouter>)
            .toJSON();

        expect(tree).toMatchSnapshot();
    })

    test("Should show settings component", async () => {
        const component = renderer
            .create(<MemoryRouter>
                <Routes>
                    <Route path="/" element={<RoomPage socket={socket.socketClient}/>} />
                </Routes>
            </MemoryRouter>)

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        let a = await component.root.findAllByProps({socket: socket.socketClient});
        expect(a[4]._fiber.elementType).toEqual(GameSettingsComponent);
    })

    test("Should show game view component", async () => {
        window.clientID = "abcd";
        window.gameData = {players: {"abcd":{
            nickname: "abc",
            color: "#FFF",
            init_ofx: 0,
            currentPiece: null,
            isImposter: false,
            hasEmergency: true,
            isExiled: false,
        }}};
        
        const component = renderer
            .create(<MemoryRouter>
                <Routes>
                    <Route path="/" element={<RoomPage socket={socket.socketClient}/>} />
                </Routes>
            </MemoryRouter>)

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        let a = await component.root.findAllByProps({socket: socket.socketClient});
        expect(a[4]._fiber.elementType).toEqual(GameViewComponent);
    })

    // test("Should copy roomid code on copy roomid button", async () => {
    //     const component = renderer
    //         .create(<MemoryRouter>
    //             <Routes>
    //                 <Route path="/" element={<RoomPage socket={socket.socketClient}/>} />
    //             </Routes>
    //         </MemoryRouter>)

    //     let tree = component.toJSON();
    //     expect(tree).toMatchSnapshot();

    //     const alertMock = jest.spyOn(window, 'alert').mockImplementation();
    //     const writeTextMocks = jest.spyOn(navigator.clipboard, 'writeText').mockImplementation();
    //     let a = await component.root.findAllByType('button');
    //     a[0].props.onClick();

    //     expect(alertMock).toHaveBeenCalled();
    //     expect(writeTextMocks).toHaveBeenCalled();
    // })
})