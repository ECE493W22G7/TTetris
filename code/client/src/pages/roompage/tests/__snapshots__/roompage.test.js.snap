// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Room Page Should show game view component 1`] = `
<div
  className="room-page"
>
  <div
    className="piece piece-1"
  />
  <div
    className="piece piece-2"
  />
  <div
    className="piece piece-3"
  />
  <div
    className="piece piece-4"
  />
  <div
    className="piece piece-5"
  />
  <div
    className="piece piece-6"
  />
  <div
    className="piece piece-7"
  />
  <div
    className="piece piece-8"
  />
  <div
    className="piece piece-9"
  />
  <div
    className="piece piece-10"
  />
  <div>
    <p
      className="h1 text-danger font-weight-bold font-italic text-center "
    >
      Treacherous Tetris
    </p>
    <link
      href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css"
      rel="stylesheet"
    />
    <a
      className=" help-button"
      href="/help"
      onClick={[Function]}
      type="button"
    >
      <i
        className="bi bi-question-circle fa-lg"
      />
    </a>
  </div>
  <div>
    <audio
      autoPlay={true}
      muted={true}
      playsInline={true}
    />
  </div>
  <div
    className="room-code"
  >
    <span
      className="h2 text-dark font-weight-bold text-center "
    >
      Room:
    </span>
    <span
      className="h2 text-dark font-weight-bold text-center"
      id="roomId"
      type="text"
      value="room id"
    />
    <button
      className="copy-button"
      onClick={[Function]}
    >
      <i
        className="bi bi-clipboard fa-lg"
      />
    </button>
  </div>
  <div
    className="room-components"
  >
    <div
      className="room-sections-left"
    >
      <div
        className="room-box-left card"
      >
        <div
          className="players-content card"
        >
          <link
            href="https://cdn.jsdelivr.net/npm/bootstrap-icons@1.7.2/font/bootstrap-icons.css"
            rel="stylesheet"
          />
          <div
            className="player-title font-italic text-center card-header"
          >
            Players Info
          </div>
          <div
            className="player-list"
          >
            <div
              className="player-info"
            >
              <div
                className="player-name-button"
              >
                <span
                  className=" text-left font-weight-bold"
                />
                <button
                  className="btn btn-link"
                  type="button"
                >
                  <span
                    className="bi bi-mic fa-lg"
                  />
                </button>
              </div>
              <div
                className="piece-container"
              >
                <div
                  className="player-piece"
                  style={
                    Object {
                      "backgroundColor": "#FFF",
                    }
                  }
                />
                <div
                  className="player-emergency"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="room-box-left card"
      >
        <div
          className="chatbox-content card"
        >
          <div
            className="chat-title font-italic text-center card-header"
          >
            Chat Box
          </div>
          <div
            className="messages-container card"
          />
          <div
            className="send-box"
          >
            <div
              className="input-group mb-3"
            >
              <input
                aria-describedby="basic-addon2"
                aria-label="Type"
                className="form-control"
                data-testid="send-box-input"
                onChange={[Function]}
                onKeyDownCapture={[Function]}
                placeholder="Type..."
                type="text"
                value=""
              />
              <div
                className="input-group-append"
              >
                <button
                  className="btn send-button"
                  onClick={[Function]}
                  type="button"
                >
                  Send
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="room-sections-right"
    >
      <div
        className="room-box-right"
      >
        <div
          className="GameViewComponent"
        >
          <div
            className="GameViewCol1"
          >
            <div>
              <div
                className="row-buttons"
              >
                <button
                  className="emergency-button"
                  onClick={[Function]}
                >
                  <i
                    className="fa fa-exclamation-circle fa-lg"
                  />
                </button>
                <div
                  className="timer-frame"
                >
                  <div
                    className="time-countdown"
                  >
                    <p>
                      0:00
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="game-level"
            >
              <p>
                Speed:
              </p>
            </div>
          </div>
          <div
            className="GameViewCol2"
            tabindex="0"
          >
            <div
              className="GameCanvas"
            >
              <div />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing Room Page Should show settings component 1`] = `
<div
  className="room-page"
>
  <div
    className="piece piece-1"
  />
  <div
    className="piece piece-2"
  />
  <div
    className="piece piece-3"
  />
  <div
    className="piece piece-4"
  />
  <div
    className="piece piece-5"
  />
  <div
    className="piece piece-6"
  />
  <div
    className="piece piece-7"
  />
  <div
    className="piece piece-8"
  />
  <div
    className="piece piece-9"
  />
  <div
    className="piece piece-10"
  />
  <div>
    <p
      className="h1 text-danger font-weight-bold font-italic text-center "
    >
      Treacherous Tetris
    </p>
    <link
      href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css"
      rel="stylesheet"
    />
    <a
      className=" help-button"
      href="/help"
      onClick={[Function]}
      type="button"
    >
      <i
        className="bi bi-question-circle fa-lg"
      />
    </a>
  </div>
  <div>
    <audio
      autoPlay={true}
      muted={true}
      playsInline={true}
    />
  </div>
  <div
    className="room-code"
  >
    <span
      className="h2 text-dark font-weight-bold text-center "
    >
      Room:
    </span>
    <span
      className="h2 text-dark font-weight-bold text-center"
      id="roomId"
      type="text"
      value="room id"
    />
    <button
      className="copy-button"
      onClick={[Function]}
    >
      <i
        className="bi bi-clipboard fa-lg"
      />
    </button>
  </div>
  <div
    className="room-components"
  >
    <div
      className="room-sections-left"
    >
      <div
        className="room-box-left card"
      >
        <div
          className="players-content card"
        >
          <link
            href="https://cdn.jsdelivr.net/npm/bootstrap-icons@1.7.2/font/bootstrap-icons.css"
            rel="stylesheet"
          />
          <div
            className="player-title font-italic text-center card-header"
          >
            Players Info
          </div>
          <div
            className="player-list"
          />
        </div>
      </div>
      <div
        className="room-box-left card"
      >
        <div
          className="chatbox-content card"
        >
          <div
            className="chat-title font-italic text-center card-header"
          >
            Chat Box
          </div>
          <div
            className="messages-container card"
          />
          <div
            className="send-box"
          >
            <div
              className="input-group mb-3"
            >
              <input
                aria-describedby="basic-addon2"
                aria-label="Type"
                className="form-control"
                data-testid="send-box-input"
                onChange={[Function]}
                onKeyDownCapture={[Function]}
                placeholder="Type..."
                type="text"
                value=""
              />
              <div
                className="input-group-append"
              >
                <button
                  className="btn send-button"
                  onClick={[Function]}
                  type="button"
                >
                  Send
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="room-sections-right"
    >
      <div
        className="room-box-right"
      >
        <div
          className="game-sections card"
        >
          <div
            className="game-settings-component card"
          >
            <div
              className="settings-content"
            >
              <span
                className="game-title font-italic text-center"
              >
                Host Settings
              </span>
              <div
                className="host-input"
              >
                <div
                  className="input-group"
                >
                  <span
                    className="input-group-text"
                    id="inputGroup-sizing-default"
                  >
                    Set Speed:
                  </span>
                  <input
                    className="form-control set-speed"
                    data-testid="speed_input"
                    disabled={true}
                    max="15"
                    min="1"
                    onChange={[Function]}
                    step="1"
                    type="range"
                    value={1}
                  />
                </div>
                <button
                  className="start-button btn btn-primary"
                  disabled={true}
                  onClick={[Function]}
                  role="button"
                  type="button"
                >
                  Start Game
                </button>
              </div>
            </div>
          </div>
          <div
            className="game-settings-component card"
          >
            <div
              className="settings-content"
            >
              <span
                className="game-title font-italic text-center"
              >
                Player Settings
              </span>
              <div
                className="player-input"
              >
                <div
                  className="input-group mb-3"
                >
                  <div
                    className="input-group-prepend"
                  >
                    <span
                      className="input-group-text"
                      id="inputGroup-sizing-default"
                    >
                      Nickname:
                    </span>
                  </div>
                  <input
                    aria-describedby="inputGroup-sizing-default"
                    aria-label="Default"
                    className="form-control"
                    data-testid="nickname_input"
                    onChange={[Function]}
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                className="player-input"
              >
                <div
                  className="input-group"
                >
                  <div
                    className="input-group-prepend"
                  >
                    <span
                      className="input-group-text"
                      id="inputGroup-sizing-default"
                    >
                      Piece Color:
                    </span>
                  </div>
                  <input
                    aria-describedby="inputGroup-sizing-default"
                    aria-label="Default"
                    className="form-control"
                    data-testid="color_input"
                    onChange={[Function]}
                    style={
                      Object {
                        "height": "40px",
                      }
                    }
                    type="color"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing Room Page Should show up 1`] = `
<div
  className="room-page"
>
  <div
    className="piece piece-1"
  />
  <div
    className="piece piece-2"
  />
  <div
    className="piece piece-3"
  />
  <div
    className="piece piece-4"
  />
  <div
    className="piece piece-5"
  />
  <div
    className="piece piece-6"
  />
  <div
    className="piece piece-7"
  />
  <div
    className="piece piece-8"
  />
  <div
    className="piece piece-9"
  />
  <div
    className="piece piece-10"
  />
  <div>
    <p
      className="h1 text-danger font-weight-bold font-italic text-center "
    >
      Treacherous Tetris
    </p>
    <link
      href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css"
      rel="stylesheet"
    />
    <a
      className=" help-button"
      href="/help"
      onClick={[Function]}
      type="button"
    >
      <i
        className="bi bi-question-circle fa-lg"
      />
    </a>
  </div>
  <div>
    <audio
      autoPlay={true}
      muted={true}
      playsInline={true}
    />
  </div>
  <div
    className="room-code"
  >
    <span
      className="h2 text-dark font-weight-bold text-center "
    >
      Room:
    </span>
    <span
      className="h2 text-dark font-weight-bold text-center"
      id="roomId"
      type="text"
      value="room id"
    />
    <button
      className="copy-button"
      onClick={[Function]}
    >
      <i
        className="bi bi-clipboard fa-lg"
      />
    </button>
  </div>
  <div
    className="room-components"
  >
    <div
      className="room-sections-left"
    >
      <div
        className="room-box-left card"
      >
        <div
          className="players-content card"
        >
          <link
            href="https://cdn.jsdelivr.net/npm/bootstrap-icons@1.7.2/font/bootstrap-icons.css"
            rel="stylesheet"
          />
          <div
            className="player-title font-italic text-center card-header"
          >
            Players Info
          </div>
          <div
            className="player-list"
          />
        </div>
      </div>
      <div
        className="room-box-left card"
      >
        <div
          className="chatbox-content card"
        >
          <div
            className="chat-title font-italic text-center card-header"
          >
            Chat Box
          </div>
          <div
            className="messages-container card"
          />
          <div
            className="send-box"
          >
            <div
              className="input-group mb-3"
            >
              <input
                aria-describedby="basic-addon2"
                aria-label="Type"
                className="form-control"
                data-testid="send-box-input"
                onChange={[Function]}
                onKeyDownCapture={[Function]}
                placeholder="Type..."
                type="text"
                value=""
              />
              <div
                className="input-group-append"
              >
                <button
                  className="btn send-button"
                  onClick={[Function]}
                  type="button"
                >
                  Send
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="room-sections-right"
    >
      <div
        className="room-box-right"
      >
        <div
          className="game-sections card"
        >
          <div
            className="game-settings-component card"
          >
            <div
              className="settings-content"
            >
              <span
                className="game-title font-italic text-center"
              >
                Host Settings
              </span>
              <div
                className="host-input"
              >
                <div
                  className="input-group"
                >
                  <span
                    className="input-group-text"
                    id="inputGroup-sizing-default"
                  >
                    Set Speed:
                  </span>
                  <input
                    className="form-control set-speed"
                    data-testid="speed_input"
                    disabled={true}
                    max="15"
                    min="1"
                    onChange={[Function]}
                    step="1"
                    type="range"
                    value={1}
                  />
                </div>
                <button
                  className="start-button btn btn-primary"
                  disabled={true}
                  onClick={[Function]}
                  role="button"
                  type="button"
                >
                  Start Game
                </button>
              </div>
            </div>
          </div>
          <div
            className="game-settings-component card"
          >
            <div
              className="settings-content"
            >
              <span
                className="game-title font-italic text-center"
              >
                Player Settings
              </span>
              <div
                className="player-input"
              >
                <div
                  className="input-group mb-3"
                >
                  <div
                    className="input-group-prepend"
                  >
                    <span
                      className="input-group-text"
                      id="inputGroup-sizing-default"
                    >
                      Nickname:
                    </span>
                  </div>
                  <input
                    aria-describedby="inputGroup-sizing-default"
                    aria-label="Default"
                    className="form-control"
                    data-testid="nickname_input"
                    onChange={[Function]}
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                className="player-input"
              >
                <div
                  className="input-group"
                >
                  <div
                    className="input-group-prepend"
                  >
                    <span
                      className="input-group-text"
                      id="inputGroup-sizing-default"
                    >
                      Piece Color:
                    </span>
                  </div>
                  <input
                    aria-describedby="inputGroup-sizing-default"
                    aria-label="Default"
                    className="form-control"
                    data-testid="color_input"
                    onChange={[Function]}
                    style={
                      Object {
                        "height": "40px",
                      }
                    }
                    type="color"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
